airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/pipeline.py

```diff
@@ -273,6 +291,22 @@ def create_connector_report(results: List[StepResult]) -> ConnectorReport:
                 metadata_upload_results = await metadata_upload_step.run()
                 results.append(metadata_upload_results)
 
+            # Try to convert the context to a PyPIPublishContext. If it returns None, it means we don't need to publish to pypi.
```

Comment:
Can you add the unit tests in this [module](https://github.com/airbytehq/airbyte/blob/main/airbyte-ci/connectors/pipelines/tests/test_publish.py#L16) to check the behavior following your addition?